From b6e46ce6b8b7b8402906af3dd4c203d3874827e2 Mon Sep 17 00:00:00 2001
From: Omair M Abdullah <omair.m.abdullah@intel.com>
Date: Fri, 18 Sep 2015 20:23:30 +0530
Subject: [PATCH 1698/2508] ASoC: dapm: fix stream directions for dsp_loopback
 links

commit 309dc8ea9a2d3ca05a4b132206b03a123cd4120e from
https://github.com/01org/linux-apollolake-i

DSP loopback links have directions reversed for the source and sink widgets.

Signed-off-by: Omair M Abdullah <omair.m.abdullah@intel.com>
---
 sound/soc/soc-dapm.c |    6 ++++--
 1 files changed, 4 insertions(+), 2 deletions(-)

diff --git a/sound/soc/soc-dapm.c b/sound/soc/soc-dapm.c
index d16c5c4..ce8b2a0 100644
--- a/sound/soc/soc-dapm.c
+++ b/sound/soc/soc-dapm.c
@@ -3516,7 +3516,8 @@ static int snd_soc_dai_link_event(struct snd_soc_dapm_widget *w,
 
 	switch (event) {
 	case SND_SOC_DAPM_PRE_PMU:
-		substream.stream = SNDRV_PCM_STREAM_CAPTURE;
+		/*substream.stream = SNDRV_PCM_STREAM_CAPTURE;*/
+		substream.stream = SNDRV_PCM_STREAM_PLAYBACK;
 		if (source->driver->ops && source->driver->ops->startup) {
 			ret = source->driver->ops->startup(&substream, source);
 			if (ret < 0) {
@@ -3530,7 +3531,8 @@ static int snd_soc_dai_link_event(struct snd_soc_dapm_widget *w,
 		if (ret < 0)
 			goto out;
 
-		substream.stream = SNDRV_PCM_STREAM_PLAYBACK;
+		/*substream.stream = SNDRV_PCM_STREAM_PLAYBACK;*/
+		substream.stream = SNDRV_PCM_STREAM_CAPTURE;
 		if (sink->driver->ops && sink->driver->ops->startup) {
 			ret = sink->driver->ops->startup(&substream, sink);
 			if (ret < 0) {
-- 
1.7.5.4

